exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/Op.kt

```diff
@@ -702,7 +703,17 @@ class QueryParameter<T>(
     /** Returns the column type of this expression. */
     val sqlType: IColumnType<T & Any>
 ) : Expression<T>() {
-    override fun toQueryBuilder(queryBuilder: QueryBuilder): Unit = queryBuilder { registerArgument(sqlType, value) }
+    internal val compositeValue: CompositeID? = (value as? EntityID<*>)?.value as? CompositeID
+
+    override fun toQueryBuilder(queryBuilder: QueryBuilder) {
+        queryBuilder {
+            compositeValue?.let {
+                it.values.entries.appendTo { (column, value) ->
+                    registerArgument(column.columnType, value)
+                }
+            } ?: registerArgument(sqlType, value)
+        }
+    }
```

Comment:
Alternatively we could add a new class like `CompositeQueryParameter` that would be very similar but might make some branches cleaner. But I couldn't think of any other applications for it so I went this way to keep the logic together.
I also didn't want to open `QueryParameter` just for an override.

Reply:
Yeah, I understand that, I tried to write something like CompositeQueryParameter also (during review of PR), and it does not look significantly better. I also tried to extend it not from `QueryParameter` but from `Expression`, what it also does not make it much better.